### ISSUE TITLE
[BUG] We need to update the GPU VMs for the tests

### ISSUE THREAD

**Issue by miguelgfierro** [recommenders-team/recommenders#1996]
### Description

Some of the GPU tests are not working. If we look in the portal, I see this error: VMSizeNotSupportedInRegion. Instances of the specified VM size are not currently available. Please retry later, try reducing the VM size or number of instances, or try deploying to a different region

**Comment by miguelgfierro** [comment]
Added new low priority nodes with VM Standard_NC8as_T4_v3 (8 cores, 56 GB RAM, 352 GB disk)

GPU nightly runs: https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036960053

**Comment by miguelgfierro** [comment]
**Standard_NC8as_T4_v3 (8 cores, 56 GB RAM, 352 GB disk), GPU Nvidia Tesla T4, Estimated cost $0.15/hr**
- ["python=3.8", group_gpu_001](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036959040#logs) succeeded 8 hours ago in 51m 25s
- ["python=3.8", group_gpu_002](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036959425#logs) succeeded 8 hours ago in 48m 12s
- ["python=3.8", group_gpu_003](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036959738#logs) succeeded 8 hours ago in 47m 39s
- ["python=3.8", group_gpu_004](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036960053#logs) succeeded 8 hours ago in 1h 11m 51s
- ["python=3.8", group_gpu_005](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036960303#logs) succeeded 8 hours ago in 57m 18s
- ["python=3.8", group_gpu_006](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036960560#logs) succeeded 8 hours ago in 1h 3m 31s
- ["python=3.8", group_gpu_007](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036960794#logs) succeeded 8 hours ago in 48m 3s
- ["python=3.9", group_gpu_001](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036961061#logs) succeeded 8 hours ago in 52m 42s
- ["python=3.9", group_gpu_002](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036961320#logs) succeeded 8 hours ago in 48m 57s
- ["python=3.9", group_gpu_003](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036961630#logs) succeeded 8 hours ago in 48m 4s
- ["python=3.9", group_gpu_004](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036961921#logs) succeeded 8 hours ago in 1h 7m 8s
- ["python=3.9", group_gpu_005](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036962401#logs) succeeded 8 hours ago in 54m 37s
- ["python=3.9", group_gpu_006](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036962804#logs) succeeded 8 hours ago in 1h 7m 48s
- ["python=3.9", group_gpu_007](https://github.com/recommenders-team/recommenders/actions/runs/6273456046/job/17036963087#logs) succeeded 8 hours ago in 49m 31s


**Standard_NC6s_v3 (6 cores, 112 GB RAM, 336 GB disk), GPU - 1 x NVIDIA Tesla V100, Estimated cost $0.61/hr per node**
- ["python=3.8", group_gpu_001](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050371080#logs) succeeded 1 hour ago in 1h 0m 29s
- ["python=3.8", group_gpu_002](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050371382#logs) succeeded 48 minutes ago in 1h 14m 10s
- ["python=3.8", group_gpu_003](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050371581#logs) succeeded 1 hour ago in 52m 57s
- ["python=3.8", group_gpu_004](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050371765#logs) succeeded 13 minutes ago in 1h 50m 25s
- ["python=3.8", group_gpu_005](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050371968#logs) succeeded 1 hour ago in 1h 5m 50s
- ["python=3.8", group_gpu_006](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050372138#logs) succeeded 40 minutes ago in 1h 26m 17s
- ["python=3.8", group_gpu_007](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050372311#logs) succeeded 1 hour ago in 1h 6m 3s
- ["python=3.9", group_gpu_001](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050372507#logs) succeeded 1 hour ago in 59m 26s
- ["python=3.9", group_gpu_002](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050372683#logs) succeeded 1 hour ago in 56m 5s
- ["python=3.9", group_gpu_003](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050372869#logs) succeeded 1 hour ago in 53m 48s
- ["python=3.9", group_gpu_004](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050373061#logs) succeeded 42 minutes ago in 1h 27m 0s
- ["python=3.9", group_gpu_005](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050373248#logs) succeeded 44 minutes ago in 1h 25m 38s
- ["python=3.9", group_gpu_006](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050373464#logs) succeeded 45 minutes ago in 1h 25m 44s
- ["python=3.9", group_gpu_007](https://github.com/recommenders-team/recommenders/actions/runs/6277833012/job/17050373618#logs) succeeded 1 hour ago in 1h 4m 37s

**Comment by miguelgfierro** [comment]
Benchmark CPU:

**Standard_A8m_v2 (8 cores, 64 GB RAM, 80 GB disk), $0.48/hr per node in dedicated, $0.08/hr in low priority**
- ["python=3.8", group_cpu_001](https://github.com/recommenders-team/recommenders/actions/runs/6273952571/job/17038434540#logs) succeeded 7 hours ago in 1h 10m 6s

**Comment by miguelgfierro** [comment]
@anargyri @SimonYansenZhao @loomlike @wutaomsft @pradnyeshjoshi @gramhagen @Leavingseason FYI the K80 machines we were using for the GPU tests have been deprecated so I had to replace them. I did a benchmrk with several options and I discovered that using the low priority VMs instead of the dedicated ones saves us 10x our costs and I don't see a difference in access. 

Some interesting discoveries:
- The T4 performs much better than the V100, even though the Vs are [4x more expensive](https://technical.city/es/video/Tesla-V100-PCIe-16-GB-vs-Tesla-T4), this might suggest that the architecture (Turing vs Volta) is more important than the power of the card. I think this is useful info if anyone is thinking of buying a card. I haven't done tests with the next generation of architectures, A100, because the options I see on Azure are only clusters of 4 GPUs.
- Looking at the CPUs, it seems that the compute optimized VMs are more efficient than general purpose and memory optimized. You might see that the comparison is not perfectly apples to apples, here I was trying to match cost first, and then capability.
- Low priority is the key. Apart from the cost, it is much easier to get cores than in dedicated, so we can scale the tests to more VMs in parallel.
